src/net.cpp

```diff
@@ -1710,34 +1710,31 @@ void ThreadMnbRequestConnections()
         std::pair<CService, std::set<uint256> > p = mnodeman.PopScheduledMnbRequestConnection();
         if(p.first == CService() || p.second.empty()) continue;
 
-        CNode* pnode = ConnectNode(CAddress(p.first), NULL, true);
-        if(pnode) {
-            grant.MoveTo(pnode->grantMasternodeOutbound);
-
-            while(true) {
-                // wait to acquire lock
-                TRY_LOCK(pnode->cs_vSend, lockSend);
-                if (!lockSend) {
-                    MilliSleep(50);
-                    continue;
-                }
+        CNode* pnode = NULL;
+        {
+            LOCK(cs_vNodes);
+            pnode = ConnectNode(CAddress(p.first), NULL, true);
+            if(!pnode) continue;
+            pnode->AddRef();
```

Comment:
This ref is held specifically for mnb purposes and it's released at the end of the loop, should be fine imo.